pandas/core/internals/concat.py

```diff
@@ -94,7 +99,7 @@ def _concatenate_array_managers(
         # concatting along the columns -> combine reindexed arrays in a single manager
         assert concat_axis == 0
         arrays = list(itertools.chain.from_iterable([mgr.arrays for mgr in mgrs]))
-        if copy:
+        if copy and axis1_needs_copy_this:
```

Comment:
might be cleaner (and avoid copies in corner cases) to do this up in the for loop right before `mgrs.append(mgr)`

Reply:
Thanks, good idea, and simplified it a bit (and just needing a dummy copy() added to NullArrayProxy)